### PR TITLE
Refactor eventHandlers

### PR DESCRIPTION
**Pull request by schaumb**
This PR does not include any changes related to the EventDispatcher refactor.

### CODE REVIEW

File: src/base/util/eventdispatcher.h [hunk 1]
```diff
@@ -30,7 +30,7 @@ class EventDispatcher
 	typedef std::shared_ptr<Event> event_ptr;
 	typedef std::function<void(Params &)> handler_fn;
 	typedef std::map<std::string, event_ptr> event_map;
-	typedef std::list<std::pair<handler_id, handler_fn>> handler_list;
+	typedef std::list<std::pair<std::uint64_t, handler_fn>> handler_list;
```
Comment (simzer) [inline src/base/util/eventdispatcher.h:33]: Do we still need handler_id then?

File: test/qtest/chart.cpp [hunk 2]
```diff
@@ -31,7 +31,8 @@ void TestChart::prepareData()
 	table.addColumn("Cat2", std::span(cat2));
 	table.addColumn("Val", std::span(val));
 
-	chart.getChart().getEventDispatcher()["pointeron"]->attach(
+	chart.getChart().getEventDispatcher().getEvent("pointeron")->attach(
+	    0,
```
Comment (simzer) [inline test/qtest/chart.cpp:35]: It seems strange that we have to pass the id form outside on the c++ interface.

**Review by simzer — CHANGES_REQUESTED** [review]
Please put a line into the changelog about the fixed bug.

File: src/base/util/eventdispatcher.h (hunk 1 above)
Comment (schaumb) [reply src/base/util/eventdispatcher.h:33]: This is part of the EventDispatcher refactor. Out of scope of this pr.

**Comment by schaumb** [comment]
This should be tested manually because no automatic tests exists.

**Review by simzer — APPROVED** [review]

**Pull request merged**